Brokerages/Alpaca/Markets/PolygonDataClient.cs

```diff
@@ -0,0 +1,260 @@
+﻿/*
+ * The official C# API client for alpaca brokerage
+ * Sourced from: https://github.com/alpacahq/alpaca-trade-api-csharp/tree/v3.5.5
+ * Changes:
+ *   * default literals to default(T)
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace QuantConnect.Brokerages.Alpaca.Markets
+{
+    /// <summary>
+    /// Provides unified type-safe access for Polygon Data API via HTTP/REST.
+    /// </summary>
+    public sealed partial class PolygonDataClient : IDisposable
+    {
+        private readonly HttpClient _httpClient = new HttpClient();
+
+        private readonly Boolean _isStagingEnvironment;
+
+        private readonly String _keyId;
+
+        /// <summary>
+        /// Creates new instance of <see cref="PolygonDataClient"/> object.
+        /// </summary>
+        /// <param name="configuration">Configuration parameters object.</param>
+        public PolygonDataClient(
+            PolygonDataClientConfiguration configuration)
+        {
+            configuration
+                .EnsureNotNull(nameof(configuration))
+                .EnsureIsValid();
+
+            _isStagingEnvironment = configuration.KeyId
+                .EndsWith("-staging", StringComparison.Ordinal);
+            _keyId = configuration.KeyId;
+
+            _httpClient.DefaultRequestHeaders.Accept
+                .Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            _httpClient.BaseAddress = configuration.ApiEndpoint;
+            _httpClient.SetSecurityProtocol();
```

Comment:
Seems this line will not be compiled unless we `#define NET45` ?
https://github.com/QuantConnect/Lean/pull/4228/files#diff-d6fdf1446f3e7eab9248163398e45ec8R141

Reply:
I define the NET45 constant at the top of the file on line 1

Reply:
It's needed in `Brokerages/Alpaca/Markets/PolygonDataClient.cs`